model/notification.go

```diff
@@ -1,12 +1,95 @@
 package model
 
+import (
+	"fmt"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
 type Notification struct {
+	// From the notifications table.
 	UserID         int    `db:"user_id"`
 	AnnouncementID int    `db:"announcement_id"`
 	Status         string `db:"status"`
 	Receipt        string `db:"receipt"`
-	ReceiptStatus  string `db:"receipt_status"`
-	Timestamp      string `db:"timestamp"`
+	// From joined tables.
+	ExpoPushToken string `db:"expo_push_token"`
+	Title         string `db:"title"`
+	Body          string `db:"body"`
 }
 
-// TODO
+func EnqueueAnnouncementNotifications(db *sqlx.DB) {
+	log.Println("Starting to enqueue announcement notifications.")
+
+	tx := db.MustBegin()
+
+	insertQuery := `
+INSERT IGNORE into notifications (user_id, announcement_id, status)
+
+SELECT users.id as user_id, announcements.id as announcement_id, "queued" as status
```

Comment:
Are these two separate queries in a single statement? Do they need a semicolon to separate them?

Reply:
It's a single query. https://dev.mysql.com/doc/refman/8.0/en/insert-select.html

Reply:
TIL, thanks.

Reply:
NP. Might be more obvious if I indented instead of having a blank line.